src/execution/radix_partitioned_hashtable.cpp

```diff
@@ -69,13 +69,28 @@ unique_ptr<GroupedAggregateHashTable> RadixPartitionedHashTable::CreateHT(Client
 //===--------------------------------------------------------------------===//
 // Sink
 //===--------------------------------------------------------------------===//
+enum class AggregatePartitionState : uint8_t {
+	//! Can be finalized
+	READY_TO_FINALIZE = 0,
+	//! Finalize is in progress
+	FINALIZE_IN_PROGRESS = 1,
+	//! Finalized, ready to scan
+	READY_TO_SCAN = 2
+};
+
 struct AggregatePartition {
 	explicit AggregatePartition(unique_ptr<TupleDataCollection> data_p)
-	    : data(std::move(data_p)), progress(0), finalized(false) {
+	    : lock(make_uniq<mutex>()), state(AggregatePartitionState::READY_TO_FINALIZE), data(std::move(data_p)),
```

Comment:
What is the reason for this being a `unique_ptr<mutex>` rather than just `mutex` ?

Reply:
Good catch. The compiler was screaming at me earlier for some reason, but I changed some stuff, and it no longer needs to be a `unique_ptr`.